### PR TITLE
add replace-persistence argument

### LINKED ISSUE #154: Proposal: Add --persist-replace argument if there is existing table
Originally we have `--persist` argument to [persist the pandas dataframe](https://jupysql.ploomber.io/en/latest/integrations/pandas.html?highlight=--persist#pandas-integration) to the target database connection.

However, if the user runs the same persist command twice, it will show the error message: `ValueError`

Take example, run this cell twice:

```
plot_data = pd.DataFrame({"x": range(0, 5), "y": range(5, 10)})
%sql --persist plot_data
```

It will produce the error message on second call:

```
ValueError: Table 'plot_data' already exists.
If you need help solving this issue, send us a message: https://ploomber.io/community
```

Which is expected, but I think what we could do better is the providing `--persist-replace` argument to do the replace-if-exists action to the table, since user might re-run the whole jupyter file several times, by using this new argument can prevent such error happens in the middle of the execution

### PR DESCRIPTION
**Pull request by Palashio**
## Describe your changes

Adds the `replace-persistence` argument in order to account for cases where dataframe already exists and user is trying to persist with the same name. Just made modifications to parameters to pass to function that already exists. 

## Issue ticket number and link
Closes #154 

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added thorough [tests](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#testing) (when necessary).
- [ ] I have added the right documentation in the [docstring](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#documenting-changes-and-new-features) and [changelog](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--244.org.readthedocs.build/en/244/

### CODE REVIEW

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/sql/magic.py [hunk 1]
```diff
@@ -391,7 +408,15 @@ def _persist_dataframe(self, raw, conn, user_ns, append=False, index=True):
         table_name = frame_name.lower()
         table_name = self.legal_sql_identifier.search(table_name).group(0)
 
-        if_exists = "append" if append else "fail"
+        if replace and append:
+            raise ValueError("You cannot both replace and append to a dataframe.")
```
Comment (tonykploomber) [inline src/sql/magic.py:412]: Let's add the test case to catch this `ValueError`!

File: src/tests/integration/test_duckDB.py [hunk 2]
```diff
@@ -17,4 +17,4 @@ def test_auto_commit_mode_off(ip_with_duckDB, capsys):
 
     # Check the tables is created
     out = ip_with_duckDB.run_cell("%sql SHOW TABLES;").result
-    assert any('weather' == table[0] for table in out)
+    assert any("weather" == table[0] for table in out)
```
[the remark this answers is not in the document]
Comment (Palashio) [reply src/tests/integration/test_duckDB.py:20]: it's just the result of using the `black` formatter.

File: src/tests/test_magic.py [hunk 3]
```diff
@@ -108,6 +108,17 @@ def test_persist(ip):
...
+def test_persist_replace(ip):
+    runsql(
+        ip, ""
+    )
+    ip.run_cell("results = %sql SELECT * FROM test;")
+    ip.run_cell("results_dframe = results.DataFrame()")
+    ip.run_cell("%sql --persist sqlite:// results_dframe")
+
+    ip.run_cell("%sql --persist-replace sqlite:// results_dframe")
+
```
[the remark this answers is not in the document]
Comment (Palashio) [reply src/tests/test_magic.py:120]: Will change it to make it more clear; but doesn't what I've written do the same thing? If the last line returns an error, the test case will error out.

**Comment by edublancas** [comment]
the docs failed but looks like it was an issue in the building system, so I re-trigger a build

**Review by tonykploomber — DISMISSED** [review]

**Review by edublancas — CHANGES_REQUESTED** [review]
pleas rebase to solve file conflicts

**Review by edublancas — CHANGES_REQUESTED** [review]
looks good. please update the docs:

here, we can show a --persist-replace example:

https://jupysql.ploomber.io/en/latest/integrations/pandas.html?highlight=persist#uploading-a-dataframe-to-the-database

here, we need to document it:

https://jupysql.ploomber.io/en/latest/api/magic-sql.html?highlight=persist#sql-sql

(release version will be 0.6.6)

**Comment by edublancas** [comment]
> @Palashio please add a changelog entry ([API Change])

this should be [Feature]  since we are adding something new, and not breaking anything

**Comment by tonykploomber** [comment]
Downgrade the sqlalchemy might fix the CI issue

**Pull request closed without merging**